spec/features/wiki/wiki_unicode_spec.rb

```diff
@@ -0,0 +1,113 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe 'Wiki unicode title spec', type: :feature, js: true do
+  let(:user) { FactoryGirl.create :admin }
+  let(:project) { FactoryGirl.create :project }
+  let(:wiki_page_1) {
+    FactoryGirl.build :wiki_page_with_content,
+                       title: '<script>alert("FOO")</script>'
+  }
+  let(:wiki_page_2) {
+    FactoryGirl.build :wiki_page_with_content,
+                       title: 'Base de données'
+  }
+  let(:wiki_page_3) {
+    FactoryGirl.build :wiki_page_with_content,
+                       title: 'Base_de_données'
+  }
+
+  let(:wiki_body) {
+    <<-EOS
+    [[Base de données]] should link to wiki_page_2
+
+    [[Base_de_données]] should link to wiki_page_2
+
+    [[base-de-donnees]] should link to wiki_page_2
+
+    [[base-de-donnees-1]] should link to wiki_page_3
```

Comment:
This is due to the same title being used in wiki_page2 and wiki_page3. The page is saved correctly, but the `acts_as_url` takes care of resolving conflicting slugs.